apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionsTable.jsx

```diff
@@ -0,0 +1,251 @@
+import React, {useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import * as Table from 'reactabular-table';
+import queryString from 'query-string';
+import Button from '@cdo/apps/templates/Button';
+import StylizedBaseDialog from '@cdo/apps/componentLibrary/StylizedBaseDialog';
+import PaginationWrapper from '@cdo/apps/templates/PaginationWrapper';
+import CloneProgrammingExpressionDialog from './CloneProgrammingExpressionDialog';
+
+const DEFAULT_VALUE = 'all';
+
+export default function ProgrammingExpressionsTable({
+  programmingEnvironmentsForSelect,
+  categoriesForSelect,
```

Comment:
This might just be me, but I found these names a little confusing (since I didn't realize "select" referred to a select tag).  I'm wondering if something like `allProgrammingEnvironments` and `allCategories` would be clearer?  Up to you.